### PR TITLE
Add info on how to use the buffers option in Postgres [ci skip]

### PR DESCRIPTION
**Pull request by boddhisattva**
### Motivation / Background

The buffer option was added as part of the Active Record Explain as part of this PR: https://github.com/rails/rails/pull/47043

However perhaps they missed updating the guides on how to use this option as part of ActiveRecord Explain specifically in the context of the PostgreSQL DB. More on how to use the Buffer option as part of PostgreSQL Explain is available as part of their official docs [here](https://www.postgresql.org/docs/current/sql-explain.html)

### Detail

This Pull Request gives a brief overview of how to use the `buffers` option with Active Record Explain using a sample query and their resulting output as a reference.

### Checklist

Before submitting the PR make sure the following are checked:

* [x] This Pull Request is related to one change. Unrelated changes should be opened in separate PRs.
* [x] Commit message has a detailed description of what changed and why. If this PR fixes a related issue include it in the commit message. Ex: `[Fix #issue-number]`
* [ ] Tests are added or updated if you fix a bug or add a feature.
* [ ] CHANGELOG files are updated for the changed libraries if there is a behavior change or additional feature. Minor bug fixes and documentation changes should not be included.

### CODE REVIEW

**Comment by boddhisattva** [comment]
I'm currently unsure why the above build is currently failing with regard to this PR and what I could do to fix it. Any suggestions please ?

**Comment by zzak** [comment]
Can you wrap columns to 80 chars?

**Comment by boddhisattva** [comment]
Thank you so much @zzak :) Your valuable suggestion helped fix the build failure :)

File: guides/source/active_record_postgresql.md [hunk 1]
```diff
@@ -15,6 +15,8 @@ After reading this guide, you will know:
 * How to implement exclusion constraints.
 * How to implement full text search with PostgreSQL.
 * How to back your Active Record models with database views.
+* How to use Explain options to gain insights that could potentially further
+  improve PostgreSQL's query performance
```
Comment (zzak) [inline guides/source/active_record_postgresql.md:19]: I'm not sure this warrants a bullet point here, this isn't meant to be a table of contents but more of an overview of the high level concepts in the guide. It's already probably a bit too specific.

Comment (boddhisattva) [reply guides/source/active_record_postgresql.md]: Happy to remove that @zzak & just did :) I was originally interpreting that each point in this section represents a part of the table of contents of the guide as a whole and I was attempting to maintain that uniformity(thus I originally added it). Also agree that it sounds more specific  👍 

P.S: My apologies for committing `wrap_text.md` by mistake and thank you so much for removing it :)

**Comment by matthewd** [comment]
This appears to be documenting the Postgres feature, not Rails / Active Record behaviour.

**Comment by boddhisattva** [comment]
@matthewd IMHO its a mix of both: A postgres feature and the related syntax of it being applied with ActiveRecord::Relation#explain syntax and hence it was intended to be added with the PostgreSQL guide specifically. 

The original intention of adding this is 
- It's a relatively less known feature that could be very useful to developers building Rails apps when one is trying to improve the performance of Rails apps with regard to the SQL query one writes via Active Record syntax.
- It could really help developers not only be more aware of it in general and it could also help give more reasons to use this in Rails app context and it touches app on Rails doing what it can to seamlessly integrate with PostgreSQL through a relevant adapter so that one could make the max use of Postgres features in Rails app context.

Does the above explanation make a reasonable case to consider including it?

**Comment by boddhisattva** [comment]
@matthewd I responded to your comment [here](https://github.com/rails/rails/pull/52893#issuecomment-2351364093), could you kindly take a look & let me know what do you think?

**Comment by boddhisattva** [comment]
@zzak I've addressed your comments, could you kindly review the PR again please?

**Comment by zzak** [comment]
Matthew makes an excellent point, and I don't think we need to go into detail explaining the output of buffers here.

I think linking to the docs to learn about it is sufficient, and there's no need to put this in the PG specific guide, how about adding it under here?

https://github.com/rails/rails/blob/193724a8e69df73388958299fd08fa6255ad4873/guides/source/active_record_querying.md#explain-options

I'm just thinking a footnote should suffice.

**Comment by boddhisattva** [comment]
@zzak Sure, I could add it under the section [here](https://github.com/rails/rails/blob/193724a8e69df73388958299fd08fa6255ad4873/guides/source/active_record_querying.md#explain-options) . My intention is to at least make people aware how to use the `buffer` option with Rails active record syntax .

Could I replace the existing example under the Explain options(which uses `Customer.where(id: 1).joins(:orders).explain(:analyze, :verbose)
` and the related Explain related output it generates with the following example:

`Company.where(id: owning_companies_ids).explain(:analyze, :verbose, :buffers)`

and related Explain SQL below ?

```
EXPLAIN (ANALYZE, VERBOSE, BUFFERS) SELECT "companies".* FROM "companies" WHERE "companies"."id" IN ($1, $2, $3) [["id", 378], ["id", 377], ["id", 373]]
                                                QUERY PLAN
-------------------------------------------------------------------------------
 Seq Scan on public.companies  (cost=0.00..2.21 rows=3 width=64) (actual time=0.010..0.012 rows=3 loops=1)
   Output: id, name, owner_id, created_at, updated_at
   Filter: (companies.id = ANY ('{378,377,373}'::bigint[]))
   Rows Removed by Filter: 10
   Buffers: shared hit=1
 Planning Time: 0.049 ms
 Execution Time: 0.024 ms
(7 rows)
```

Could the above updated example be good to go with?

**Comment by zzak** [comment]
Sorry I don't think it's necessary here, we don't need to document every option postgres supports.

Thanks for your PR.

**Comment by boddhisattva** [comment]
@zzak I would like to start with thanking you for your valuable time to collaborate with me on this PR and for discussing various related options. This way I feel seen and heard and hence I sincerely appreciate it more.

> we don't need to document every option postgres supports.

I agree with you on this in general. 

> Sorry I don't think it's necessary here

**If I can take a leap of faith and just attempt to be fully honest with you**, I honestly feel confused about a few things at this point after reading the above and this confusion is because of a couple of reasons:

1. My confusion mainly stems from that although the Postgres guide for ActiveRecord [here](https://guides.rubyonrails.org/active_record_postgresql.html) originally mentions it is meant to cover PostgreSQL specific usage(from what I read at the top of the guide especially which says: `This guide covers PostgreSQL specific usage of Active Record.`), I currently feel this guideline: **doesn't seem to clearly specify what can and cannot be included and it seems to be up for one's interpretation**. What do you think about this ? Do you think it could be worth improving this guideline at least if you also think it can be made clearer for one's future reference while attempting to add more contributions in general going forward with regard to this guide?

2. The above guideline was a key reason why I initially added the `buffer` option in the Postgres guide for active record as it felt like it goes inline with what the guide stood for. Am I missing something in how I'm interpreting this line(i.e., `This guide covers PostgreSQL specific usage of Active Record.`)?


3. Your comment [here](https://github.com/rails/rails/pull/52893#issuecomment-2359904220) seemed like you were ok with adding it in the other section [here](https://github.com/rails/rails/blob/193724a8e69df73388958299fd08fa6255ad4873/guides/source/active_record_querying.md#explain-options) as a foot note.  Now you feel we might not need it at all.

4. Lastly, people have a notion along the lines of Rails is not good enough for apps at scale when it comes to its performance and Rails contributors and others who believe in what Rails can do in general as a framework are taking steps to counter that by adding what they can to make Rails more performant, directly or indirectly. An attempt to add this documentation was originally with the intention to support that Rails can do so much more to support apps at scale directly or indirectly. This is one more reason which motivated me to add this documentation in the first place. 

At the end of the day, I respect your and the teams decision if you still think this might not be worth adding, I would like to request to at least bring up for your kind consideration if the [Postgres guide for ActiveRecord](https://guides.rubyonrails.org/active_record_postgresql.html) guideline on top of the guide could be improved in the spirit of giving people more clarity of what can be potentially added to this guide especially for those who'd like to contribute to this specific guide going forward.

Thank you again for your valuable time :) 🙏

**Comment by zzak** [comment]
For the Postgres guide, I think let's compare something like [unique constraints](https://edgeguides.rubyonrails.org/active_record_postgresql.html#unique-constraint). Something that is extremely important to running the database and yet it is only 3 full sentences and a small code sample.

Let me suggest something more concrete.

* Postgres Guide could have a link to explain options in the query guide
* Add a note about buffers with a link to the pg docs
* Don't replace existing code samples

If this was something that application really depends on, or is used almost every day, than I'd say definitely it should go in. But I don't feel strongly about it.

**Comment by boddhisattva** [comment]
@zzak Thank you for your feedback and for your understanding :) I've updated the docs as per your suggestions. Could you kindly review this again please ?

**Comment by zzak** [comment]
@boddhisattva I've force pushed a commit with my suggestion, let's wait for another review. Thanks for your patience.

@p8 what do you think?

**Comment by boddhisattva** [comment]
@zzak Sure, thank you.

**Comment by boddhisattva** [comment]
Thank you very much for merging these changes @byroot :) 🙏  Also thank you very much @zzak for all the collaboration we had as part of this PR, for your valuable time & for hearing me out :) 🙏

**Pull request merged**